hash_practice/exercises.py

```diff
@@ -1,29 +1,115 @@
 
+from attr import has
+
+
 def grouped_anagrams(strings):
-    """ This method will return an array of arrays.
+    """ 
+    - Use a hash table to solve a coding problem
+    - Identify how a hash table can produce a more attractive runtime over alternative solutions
+    Given an array of strings, group anagrams together.
+    ### Example
+    ```
+    Input: ["eat", "tea", "tan", "ate", "nat", "bat"],
+    Output:
+    [
+    ["ate","eat","tea"],
+    ["nat","tan"],
+    ["bat"]
+    ]
+    ```
+    Note:
+    - All inputs will be in lowercase.
+    - The order of your output does not matter
+
+        This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        
+        Time Complexity: O(m * nlogn) - | python sorting algorithm -> O(n. logn).

```

Comment:
While this is an accurate answer assuming `n` stands for the number of characters in the strings and `m` stands for the number of strings in the list, we can make a simplifying assumption. Since we know the strings are English words, we can be pretty sure they will not get too big in size. On average words in English have about 5 letters, which is going to very quickly get dwarfed by the number of words in the list so we can simply say this is O(m) in time complexity.